src/goto-symex/symex_assign.cpp

```diff
@@ -63,7 +63,7 @@ void goto_symext::symex_assign(
       statement == ID_cpp_new || statement == ID_cpp_new_array ||
       statement == ID_java_new_array_data)
       symex_cpp_new(state, lhs, side_effect_expr);
-    else if(statement==ID_allocate)
+    else if(statement==ID_allocate || statement==ID_java_new)
```

Comment:
How come? This should have been removed by `remove_java_new`!

Reply:
Previously remove_java_new could not be run per-function, because it lowered into a sized malloc, and types could later gain fields when method conversion encountered a field access against a stubbed type. However this is no longer true, so the whole remove_java_new malarkey can now go away again. I'll submit a PR to that effect on Monday.